src/imitation/util/sacred_file_parsing.py

```diff
@@ -0,0 +1,70 @@
+"""Utilities to parse sacred run directories."""
+import json
+import pathlib
+import warnings
+from collections import defaultdict
+from typing import Any, Dict, Generator, List, Tuple
+
+SacredRun = Dict[str, Any]
+SacredConfAndRun = Tuple[Dict[str, Any], SacredRun]
+GroupedRuns = Dict[str, Dict[str, List[SacredRun]]]
+
+
+def find_sacred_runs(
+    run_path: pathlib.Path,
+    only_completed_runs: bool = False,
+) -> Generator[SacredConfAndRun, None, None]:
+    """Recursively iterates the sacred runs found below the given path.
+
+    Assumes runs in the format of the sacred FileStorageObserver: each run consists
+    of a folder that contains a config.json and a run.json file.
+
+    Note: will work with nested directories and can therefore be applied to the
+    `output/sacred` folder of the command line interface which creates sub-folders for
+    each script.
+
+    Args:
+        run_path: The path to search for sacred run directories.
+        only_completed_runs: If True, only yields runs that have a run.json file with
+            status "COMPLETED".
+
+    Yields:
+        Tuples of (config, run) dicts.
+    """
+    for config_path in run_path.rglob("config.json"):
+        run_path = config_path.parent / "run.json"
+
+        if run_path.exists():
+            run = json.loads(run_path.read_text())
+            if only_completed_runs and run["status"] != "COMPLETED":
```

Comment:
I decided against the warning here since IMO the `only_completed_runs==True` makes it clear that we are not interested in failed/pending runs.
The summary will contain a special section listing the number of failed/pending/running runs if there are any.